### PR TITLE
upgrade go to 1.22.2

### PR DESCRIPTION
**Pull request by AdamHaffar**
### Description

Upgrades Go version to 1.22.2 and updates GH actions to use go.mod instead of a hard-coded Go version.

Fixes #998

### Quick checks:

- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-databricks/pulls) for the same update/change.
- [ ] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: repository/snowflake.go [hunk 1]
```diff
@@ -180,6 +180,9 @@ func (s *Snowflake) GetTrackingData(
 	}
 
 	rows, err := tx.QueryContext(ctx, buildGetTrackingData(trackingTable, fields, offset, limit))
+	if rows.Err() != nil {
+		return nil, fmt.Errorf("run query: %w", rows.Err())
+	}
```
Comment (hariso) [inline repository/snowflake.go:185]: I guess a linter complained? Anyway, at this point, `rows` can be nil (if `err` is not nil), so rows.Err() would panic in that case. This check can be moved below the err != nill check.

File: repository/snowflake.go [hunk 2]
```diff
@@ -237,6 +244,9 @@ func (s *Snowflake) GetMaxValue(ctx context.Context, table, orderingColumn strin
 	if err != nil {
 		return nil, fmt.Errorf("query get max value: %w", err)
 	}
+	if rows.Err() != nil {
+		return nil, fmt.Errorf("query get max value: %w", err)
+	}
```
Comment (lovromazgon) [inline repository/snowflake.go:249]: We actually need to move these checks after calling `rows.Next()`, this error tells you if/what went wrong when iterating through the result set (see [Errors From Iterating Resultsets](http://go-database-sql.org/errors.html#errors-from-iterating-resultsets)).

File: repository/snowflake.go [hunk 3]
```diff
@@ -183,6 +181,9 @@ func (s *Snowflake) GetTrackingData(
 	if err != nil {
 		return nil, fmt.Errorf("run query: %w", err)
 	}
+	if rows.Err() != nil {
+		return nil, fmt.Errorf("run query: %w", rows.Err())
+	}
```
Comment (lovromazgon) [inline repository/snowflake.go:186]: We also need to move this check...

File: repository/snowflake.go [hunk 4]
```diff
@@ -226,6 +227,10 @@ func (s *Snowflake) TableExists(ctx context.Context, table string) (bool, error)
 	if err != nil {
 		return false, err
 	}
+	if rows.Err() != nil {
+		return false, fmt.Errorf("table exists: %w", err)
+	}
+
```
Comment (lovromazgon) [inline repository/snowflake.go:233]: ... and this one after `rows.Next()`.

Comment (lovromazgon) [reply repository/snowflake.go:238]: Definitely `if err = rows.Err(); err != nil`, otherwise `err` will be `nil`.

**Pull request merged**